### PR TITLE
Longhorn unset node

### PR DESCRIPTION
**Pull request by mattmattox**
Workaround for https://github.com/longhorn/longhorn/issues/2618

### CODE REVIEW

File: longhorn/unset-nodeID/main.sh [hunk 1]
```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
```
[the remark this answers is not in the document]
Comment (mattmattox) [reply longhorn/unset-nodeID/main.sh]: Updated

File: longhorn/unset-nodeID/main.sh [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+if [[ -z $LoopSleep ]]
```
[the remark this answers is not in the document]
Comment (mattmattox) [reply longhorn/unset-nodeID/main.sh]: Done

File: longhorn/unset-nodeID/main.sh [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+fi
+if [[ -z $AlertOnly ]]
+then
+  AlertOnly="False"  
+fi
+
+while true
+do
+
+for volume in `kubectl -n $LonghornNamespace get volumes.longhorn.io -o NAME| awk -F'/' '{print $2}'`
```
[the remark this answers is not in the document]
Comment (mattmattox) [reply longhorn/unset-nodeID/main.sh]: Done

File: longhorn/unset-nodeID/main.sh [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+          fi
+        else
+          echo "CRITICAL: $volume failed to patch"
+        fi
+      fi
+    fi
+  fi
+done
+
+if [[ ! -z $Debug ]]; then echo "Sleeping..."; fi
```
[the remark this answers is not in the document]
Comment (mattmattox) [reply longhorn/unset-nodeID/main.sh]: I have it sleeping to stop the script from hammering the kube-apiserver. You can set the Sleep to 0 to skip the sleep.

**Review by oxr463 — COMMENTED** [review]
Personally, I would change ALERT_ONLY and DEBUG, but it looks good to me.

File: longhorn/unset-nodeID/main.sh (hunk 4 above)
[the remark this answers is not in the document]
Comment (oxr463) [reply longhorn/unset-nodeID/main.sh]: I would put a comment in the script and/or in the README.md file for future reference.

File: longhorn/unset-nodeID/workload.yaml [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+      serviceAccount: unset-nodeid
+      serviceAccountName: unset-nodeid
+      containers:
+      - args:
+        - /root/bin/longhorn_unset_node_id.sh
+        env:
+        - name: LOOP_SLEEP
+          value: "60"
+        - name: LONGHORN_NAMESPACE
+          value: "longhorns-system"
```
Comment (dkeightley) [inline longhorn/unset-nodeID/workload.yaml:57]:
```suggestion
          value: "longhorn-system"
```

**Review by oxr463 — APPROVED** [review]

**Pull request merged**